### PR TITLE
CardReader: allow direct upload of Files

### PR DESCRIPTION
**Pull request by ParisMi**
allows upload of cards directly from a `File`.

### CODE REVIEW

**Comment by MattSainz** [comment]
LGTM

File: src/main/java/com/fullcontact/api/libs/fullcontact4j/http/cardreader/CardReaderUploadRequest.java [hunk 1]
```diff
@@ -1,56 +1,97 @@
...
-        private String cardBack;
 
         public Builder verified(CardReaderQuality quality) {
             params.put(FCConstants.PARAM_CARD_VERIFIED, quality.name());
             return this;
         }
 
         public Builder cardFront(InputStream stream) {
-            cardFront = Utils.encodeToStringAndClose(stream);
+            cardFrontStr = Utils.encodeToStringAndClose(stream);
```
Comment (jimhooker2002) [inline src/main/java/com/fullcontact/api/libs/fullcontact4j/http/cardreader/CardReaderUploadRequest.java]: encodeToStringAndClose is the method I was originally concerned with.  This turns the stream into a string in memory.   I'm hoping there's a way to send the stream through Retrofit without turning it into a String.  The String inhales the file into memory which is a potential scaling problem.  Really I'd like the library to work with the stream throughout.  Wanted cardFront to take an InputStream which is then streamed through Retrofit and onto FullContact - all without inhaling stream into a String in memory.

**Comment by jentfoo** [comment]
@ParisMi I just pushed a commit that should address @jimhooker2002 's concerns.  I have not tested it beyond the unit tests, but besides the newly added `TypedInputStream` none of my changes are earth shattering.  Would you please look them over and test before merging?

**Comment by jimhooker2002** [comment]
Thanks again for the fast response on this.  The latest changes look good to me.  Streaming goodness all the way.  Love it!

**Comment by jimhooker2002** [comment]
Sorry to chase.  Do you have a rough ETA when this will be merged to master and released?

**Comment by ParisMi** [comment]
I'll work on that today and mention you here when it's out.

**Comment by ParisMi** [comment]
version `5.4.0` released @jimhooker2002

**Pull request merged**